### PR TITLE
default thread current step to base of tree if not supplied

### PR DESCRIPTION
**Pull request by Flip-Liquid**
Fixes PRO-647

### CODE REVIEW

File: bot/common/commands.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ async def add_dao(ctx: discord.ApplicationContext):
 #                return
 #            thread = await GuildSelect(
 #                ctx.author.id,
-#                hashlib.sha256("".encode()).hexdigest(),
+#                None,
 #                message.id,
 #                "",
 #            )
```
Comment (alexkeating) [inline bot/common/commands.py:347]: We can probably remove this commented out step right?

File: bot/common/threads/thread_builder.py [hunk 2]
```diff
@@ -198,7 +196,14 @@ def __await__(self):
 
     async def _init_steps(self):
         self.steps = await self.get_steps()
-        self.step = self.find_step(self.steps, self.current_step)
+        # assign the current_step hash to root of the tree if no overriding
+        # current_step is provided when constructing the thread
+        if self.current_step is None:
+            self.current_step = self.steps.hash_
```
Comment (alexkeating) [inline bot/common/threads/thread_builder.py:202]: Why do we need to change this now? Is it related to this change? https://github.com/Govrn-HQ/KevinMalone/pull/160/files#diff-8c706a8372a231a6d172054311fb86d484f30a2cdd2ebb7cc13a48a02c562d3fR16

**Review by alexkeating — COMMENTED** [review]
Looks good, I have a couple of questions and the tests need to be fixed

File: bot/common/commands.py (hunk 1 above)
Comment (Flip-Liquid) [reply bot/common/commands.py:347]: yep we def can.

File: bot/common/threads/thread_builder.py (hunk 2 above)
Comment (Flip-Liquid) [reply bot/common/threads/thread_builder.py:202]: yep, it's related to that change. missed something in that PR, if you create a thread and immediately send it, it runs into a snag.-

**Pull request merged**